### PR TITLE
#393 - Feature: pass in the page when emitting the page change event

### LINKED ISSUE #393: Feature: pass in the page when emitting the page change event
**Feature: What is your use case for such a feature?**

Right now when the page change event is emitted it's empty. it'll be nice to get the page it was changed to passed in.

This might not be an actual feature or a bug but rather a minor change. I can submit a PR if needed. ~~I was also wondering if there is a way to access the current page in any other way?~~

Edit: I guess the page can be accessed with the following code:

`this.$refs.search._localSearchStore.page`

However the page being passed as part of the event is helpful.

Thank You.

### PR DESCRIPTION
**Pull request by VinceG**
fixes #393

### CODE REVIEW

**Review by Haroenv — APPROVED** [review]
Thank you! Since people didn't expect anything in that callback, we can release this as a patch I think

**Comment by Haroenv** [comment]
Can you edit the test too? 

https://github.com/algolia/vue-instantsearch/blob/40c9406c5f0d477324a2e9061c75241f6debad6e/src/components/__tests__/pagination.js#L93-L118

**Comment by VinceG** [comment]
@Haroenv Done.

File: src/components/__tests__/pagination.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ test('it should emit a "page-change" event when page changes', () => {
 
   vm.$mount();
 
-  expect(onPageChange).not.toHaveBeenCalled();
+  expect(onPageChange).not.toHaveBeenCalledWith('page');
```
Comment (Haroenv) [inline src/components/__tests__/pagination.js]: This should have stayed the same, it's testing that it wasn't called yet

File: src/components/__tests__/pagination.js [hunk 2]
```diff
@@ -108,7 +108,7 @@ test('it should emit a "page-change" event when page changes', () => {
...
 
   vm.$el
     .getElementsByTagName('li')[3]
```
Comment (Haroenv) [inline src/components/__tests__/pagination.js]: after these lines add: 


```js
expect(onPageChange).toHaveBeenCalledWith(page);
```

**Review by Haroenv — CHANGES_REQUESTED** [review]
Thanks for adding the tests, I think you changed the wrong test though.

File: src/components/__tests__/pagination.js (hunk 2 above)
Comment (VinceG) [reply src/components/__tests__/pagination.js]: @Haroenv  How about this:

```
test('it should emit a "page-change" event when page changes and pass in the page variable', () => {
  const searchStore = {
    page: 1,
    totalPages: 20,
    totalResults: 4000,
  };
  const Component = Vue.extend(Pagination);
  const vm = new Component({
    propsData: {
      searchStore,
    },
  });

  const onPageChange = jest.fn();
  vm.$on('page-change', onPageChange);

  vm.$mount();

  expect(onPageChange).not.toHaveBeenCalled();

  vm.$el
    .getElementsByTagName('li')[3]
    .getElementsByTagName('a')[0]
    .click();
  expect(onPageChange).toHaveBeenCalledTimes(1);

  const page = searchStore.page;
  expect(onPageChange).toHaveBeenCalledWith(page);
  expect(page).toBe(2);
});
```

Comment (Haroenv) [reply src/components/__tests__/pagination.js]: Exactly what I meant!

**Review by Haroenv — APPROVED** [review]
Great! Thanks for your contribution. I’ll release this soon

**Pull request merged**